src/ensenso.cpp

```
@@ -969,4 +969,40 @@ Result<Ensenso::CaptureParams> Ensenso::getCaptureParameters(bool crop_to_roi) {
 	return params;
 }
 
+bool Ensenso::isDumpingTree() {
+	return dump_tree_;
+}
+
+void Ensenso::enableNxLibLogging(std::string connection_folder_path, std::string debug_level, int log_file_size, bool dump_tree) {
+	NxLibItem debugOut = root[itmDebug][itmFileOutput];
+	debugOut[itmFolderPath] = connection_folder_path;
+	debugOut[itmMaxTotalSize] = log_file_size;
+	debugOut[itmEnabled] = true;
+	root[itmDebug][itmLevel] = debug_level;
+	dump_tree_ = dump_tree;
+	connection_folder_path_ = connection_folder_path;
+}
+
+void Ensenso::dumpTree(std::string time_stamp) {
```

I would like this to return an empty result rather than doing the logging itself.

Let's return the error messages instead of logging them, and the caller can decide what to do with them.